bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,312 @@
+                        continue;
+                    }
+
+                    @Nullable
+                    ThingHandler handler = thing.getHandler();
+                    if (handler == null) {
+                        logger.debug("No handler for thing {} refresh cancelled", thing.getUID());
+                        continue;
+                    }
+                    thing.getChannels().forEach(channel -> handler.handleCommand(channel.getUID(), REFRESH));
+                } catch (Exception ex) {
+                    logger.warn("Cannot refresh thing {} from CloudBridgeHandler", thing.getUID(), ex);
+                }
+            }
+
+            // all things were updated,
+            updateStatus(ONLINE);
+        } catch (Exception ex) {
+            logger.warn("Cannot refresh devices from CloudBridgeHandler", ex);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands in this bridge
+        logger.debug("Bridge does not support any commands to any channels. channelUID={}, command={}", channelUID,
+                command);
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> localScheduledFuture = scheduledFuture;
+        if (localScheduledFuture != null) {
+            localScheduledFuture.cancel(true);
+            scheduledFuture = null;
+        }
+        super.dispose();
+    }
+
+    @Override
+    public SortedSet<@NonNull DeviceProperty<?>> findPropertiesForDevice(String dsn) {
+        return requireNonNullElse(devicePropertiesCache.get(dsn), emptySortedSet());
+    }
+
+    @Nullable
+    private SortedSet<DeviceProperty<?>> loadPropertiesForDevice(String dsn)
+            throws ExecutionException, InterruptedException, TimeoutException {
+        @Nullable
+        SalusApi api = salusApi;
+        if (api == null) {
+            logger.debug("Cannot find properties for device {} because salusClient is null", dsn);
+            return null;
+        }
+        logger.debug("Finding properties for device {} using salusClient", dsn);
+        ApiResponse<SortedSet<DeviceProperty<?>>> response = api.findDeviceProperties(dsn);
+        if (response.failed()) {
+            var error = response.error();
+            if (error == null) {
+                error = new Error(500, "Unknown error");
+            }
+            throw new RuntimeException("Cannot find properties for device " + dsn + " using salusClient\n"
+                    + error.code() + ": " + error.message());
+        }
+        return response.body();
+    }
+
+    @Override
+    public boolean setValueForProperty(String dsn, String propertyName, Object value) {
+        try {
+            @Nullable
+            SalusApi api = salusApi;
+            if (api == null) {
+                logger.warn("Cannot set value for property {} on device {} because salusClient is null", propertyName,
```

Comment:
I think the thingstatus should be set to offline here as the binding does not apear to recover from this. Although somewhat theoretically

Reply:
Technically `salusApi` cannot be null (set in `init` and unset in `dispose`). I will just add `requireNonNull`

Reply:
ok